staging/src/k8s.io/apimachinery/pkg/api/resource/amount.go

```diff
@@ -203,6 +203,50 @@ func (a *int64Amount) Sub(b int64Amount) bool {
 	return a.Add(int64Amount{value: -b.value, scale: b.scale})
 }
 
+// Mul multiplies two int64Amounts together, matching scales.
+// It will return false and not mutate a if overflow or underflow would result.
+// Mul adds the value of a to itself n times.
+// It will return false and not mutate a if overflow or underflow would result.
+func (a *int64Amount) Mul(b int64Amount) bool {
+	switch {
+	case b.value == 0:
+		a.value = 0
+		return true
+	case a.value == 0:
+		a.value = 0
+		a.scale = b.scale
+		return true
+	case a.scale == b.scale:
+		c, ok := int64Multiply(a.value, b.value)
+		if !ok {
+			return false
+		}
+		a.value = c
+	case a.scale > b.scale:
```

Comment:
This isn't the thing to do at all (see my other comment), but for future reference, the natural way to multiply two numbers in the form a * b ^ c is just a1*a2 * b^(c1+c2). That is, values multiply and scales add. Great care needs to be taken when values are large.

Reply:
Good explanation. Thanks, @lavalamp .

"... the natural way to multiply two numbers in the form a * b ^ c is just a1*a2 * b^(c1+c2). That is, values multiply and scales add. Great care needs to be taken when values are large.
"